python/foglamp/common/web/tls_helper.py

```diff
@@ -0,0 +1,146 @@
+# -*- coding: utf-8 -*-
+
+# FOGLAMP_BEGIN
+# See: http://foglamp.readthedocs.io/
+# FOGLAMP_END
+
+from OpenSSL.crypto import (
+    FILETYPE_ASN1, X509Store, X509StoreContext, X509StoreContextError,
+    load_certificate)
+from base64 import b64decode
+from foglamp.common import logger
+
+__author__ = "Amarendra Kumar Sinha"
+__copyright__ = "Copyright (c) 2017 OSIsoft, LLC"
+__license__ = "Apache 2.0"
+__version__ = "${VERSION}"
+
+_logger = logger.setup(__name__)
+
+
+class VerificationError(ValueError):
+    pass
+
+
+class BaseCert:
+    @classmethod
+    def from_pem(cls, pem_data):
+        try:
+            if not isinstance(pem_data, str):
+                raise AssertionError("pem_data is not string: %s", pem_data)
```

Comment:
not sure `AssertionError` will be stripped? if we use python -O or -OO flag, as for the assert statement/s

Reply:
AssertionError extends Exception class and as such it will NEVER be hidden by -O or -OO flags.